libsawtooth/src/client/rest.rs

```diff
@@ -59,14 +61,29 @@ impl RestApiSawtoothClientBuilder {
         self.url = Some(url.into());
         self
     }
+    /// Sets auth
+    ///
+    /// # Arguments
+    ///
+    /// * `auth` - The credentials of the request authorizer.
+    pub fn with_basic_auth(mut self, username: &str, password: &str) -> Self {
+        self.auth = Some((username.into(), password.into()));
+        self
+    }
     /// Builds a RestApiSawtoothClient
     ///
     /// Returns an error if the url is not set
     pub fn build(self) -> Result<RestApiSawtoothClient, SawtoothClientError> {
+        let auth = if let Some(auth) = self.auth {
```

Comment:
`self.auth.map(..)` is better for this situation because we only need to transform the contents of a `Some`.

Reply:
Fixed